### DISCUSSION TITLE
Including markdown files that are stored in another directory

### DISCUSSION THREAD

**Discussion by LisaHopcroft** [quarto-dev/quarto-cli#2315] [Q&A] [answered]
Caveat: apologies in advance if I've missed an existing discussion, I have tried to find one!!

I have a library of markdown content that I'd like to pull into a Quarto presentation; this library exists in a different directory to the quarto presentation.

I know that I can include the absolute/relative path to the file, but I would rather set the directory using a variable.

Is there a way to either:
1. set a `$PATH` type variable that Quarto can use to find markdown files that are referred to in a `.qmd` file (e.g., in an `{{< include XXX >}}` statement?
2. pass a variable (containing the location) to be used in an include statement? Something like `{{< include X/_info.qmd >}}`

**Comment by cscheid** [comment, marked as answer]
No, there isn't. As you've noticed, the `include` functionality is pretty narrow. That's currently by design, because the semantics of inclusions are very confusing.  Quarto used to do something only slightly more complex in terms of directory resolution and we had to revert the change because the behavior was very unintuitive.

<strike>There is a general technique you can use, and that is to use root-relative ("absolute") paths, starting with a `/`. If you write `{{< include /path/to/include.md >}}` then that path gets resolved relative to the project website, independently of the path of the current file. This allows you to create a subdirectory of useful snippets and then refer to them consistently.</strike> EDIT: There's a bug in quarto where this isn't working properly. I'm fixing it right away.

**Reply by cscheid** [reply]
Bugfix incoming on https://github.com/quarto-dev/quarto-cli/pull/2321. It should be available on tonight's pre-release builds.
